frontend/views/components/Modal/Modal.vue

```diff
@@ -5,74 +5,95 @@
-      // What are the use cases of it?
-      content: null,
-      subcontent: []
+      content: null, // This is the main modal
+      subcontent: [] // This is for collection of modal on top of modals
     }
   },
   created () {
-    sbp('okTurtles.events/on', LOAD_MODAL, component => this.openModal(component))
-    sbp('okTurtles.events/on', UNLOAD_MODAL, component => this.closeModal(component))
+    sbp('okTurtles.events/on', OPEN_MODAL, component => this.openModal(component))
+    sbp('okTurtles.events/on', UNLOAD_MODAL, component => this.unloadModal(component))
     sbp('okTurtles.events/on', REPLACE_MODAL, component => this.replaceModal(component))
+    // When press escape it should close the modal
+    keyboardEvent = window.addEventListener('keyup', (e) => {
+      // Only if there an active modal
+      if (this.content && e.key === 'Escape') {
+        this.unloadModal()
+      }
+    })
   },
   mounted () {
-    const modal = this.$route.query.modal
-    if (modal) this.openModal(modal)
+    this.initializeModals()
   },
   beforeDestroy () {
-    sbp('okTurtles.events/off', LOAD_MODAL, this.openModal)
-    sbp('okTurtles.events/off', UNLOAD_MODAL, this.closeModal)
-    sbp('okTurtles.events/off', REPLACE_MODAL, this.replaceModal)
+    sbp('okTurtles.events/off', OPEN_MODAL)
+    sbp('okTurtles.events/off', UNLOAD_MODAL)
+    sbp('okTurtles.events/off', REPLACE_MODAL)
+    window.removeEventListener('keyup', keyboardEvent)
   },
   watch: {
     '$route' (to, from) {
-      if (from.query.modal && !to.query.modal) {
-        sbp('okTurtles.events/emit', CLOSE_MODAL)
+      if (to.query.modal) {
+        // We reset the modals with no animation for simplicity
+        if (to.query.modal !== this.content) this.content = to.query.modal
+        const subcontent = to.query.subcontent
+        if (subcontent && subcontent !== this.activeSubcontent()) {
+          // Try to find the new subcontent in the list of subcontent
+          const i = this.subcontent.indexOf(subcontent)
+          if (i) this.subcontent = this.subcontent.splice(0, i)
+          else this.subcontent = subcontent
+        }
+      } else {
+        // When the route change we compare to see if the modal changed
+        // If so, we send the event to close the modal
+        if (from.query.modal) {
+          sbp('okTurtles.events/emit', CLOSE_MODAL)
+        }
       }
     }
   },
   methods: {
+    activeSubcontent () {
+      return this.subcontent[this.subcontent.length - 1]
+    },
+    initializeModals () {
+      const modal = this.$route.query.modal
+      if (modal) this.openModal(modal)
+      const subcontent = this.$route.query.subcontent
+      if (subcontent) this.openModal(subcontent)
+    },
+    updateUrl () {
+      if (this.content) {
+        this.$router.push({ query: { modal: this.content, subcontent: this.activeSubcontent() } })
+      } else {
+        this.$router.push({})
```

Comment:
I believe we shouldn't use `push({})`, because this removes all the queries from an URL. For example:

```
http://localhost:8000/app/?id=special&modal=SignUp
```

when the modal is closed, the other queries are also removed

```
http://localhost:8000/app/
```

And maybe we might want to keep the other queries... I can't point any existing example at the moment but in other projects it's a common constraint I had to handle, so I just wanted to remind you of that.